arduino-ide-extension/src/browser/contributions/add-zip-library.ts

```diff
@@ -49,21 +50,60 @@ export class AddZipLibrary extends SketchContribution {
         });
         if (!canceled && filePaths.length) {
             const zipUri = await this.fileSystemExt.getUri(filePaths[0]);
-            const dialog = new InstallationProgressDialog('Installing library', 'zip');
             try {
-                this.outputChannelManager.getChannel('Arduino').clear();
-                dialog.open();
-                await this.libraryService.installZip({ zipUri });
-            } catch (e) {
-                this.messageService.error(e.toString());
-            } finally {
-                dialog.close();
+                await this.doInstall(zipUri);
+            } catch (error) {
+                if (error instanceof AlreadyInstalledError) {
+                    const result = await new ConfirmDialog({
+                        msg: error.message,
+                        title: 'Do you want to overwrite the existing library?',
+                        ok: 'Yes',
+                        cancel: 'No'
+                    }).open();
+                    if (result) {
+                        await this.doInstall(zipUri, true);
+                    }
+                }
+            }
+        }
+    }
+
+    private async doInstall(zipUri: string, overwrite?: boolean): Promise<void> {
+        const dialog = new InstallationProgressDialog('Installing library', 'zip');
+        try {
+            this.outputChannelManager.getChannel('Arduino').clear();
+            dialog.open();
+            await this.libraryService.installZip({ zipUri, overwrite });
+        } catch (error) {
+            if (error instanceof Error) {
+                const match = error.message.match(/library (.*?) already installed/);
+                if (match && match.length >= 2) {
+                    const name = match[1].trim();
+                    if (name) {
+                        throw new AlreadyInstalledError(`A library named ${name} already exists. Do you want to overwrite it?`, name);
```

Comment:
```suggestion
                        throw new AlreadyInstalledError(`A library folder named ${name} already exists. Do you want to overwrite it?`, name);
```
The library name is defined by the `name` field of `library.properties`, but this is about a folder name collision (two libraries that use the same installation folder name could have completely different `name` values).

I realize that this is following the convention established by the Arduino CLI error message, but since the message text is defined here in the IDE code base, I think it's worth fixing it here from the start.